dev-docs/RFCs/v7.x/python-integration-rfc.md

```diff
@@ -0,0 +1,117 @@
+# RFC: Jupyter Notebook/Python Integration Strategy for deck.gl and applications
+
+* **Author**: Ib Green, Andrew Duberstein
+* **Date**: Mar, 2019
+* **Status**: **Draft**
+
+## Summary
+
+This RFC proposes adding a minimal official Python integration to deck.gl, focusing on Jupyter notebook support.
+
+
+## Background
+
+Improved support for data science and machine learning visualization is a focus area for deck.gl, and deck.gl is already very useful for geospatial visualizations in these contexts. Many data scientists work in Python and being able to effortlessly visualize data generated by libraries like numpy and pandas directly from the notebook is very valuable.
+
+By providing a set of very simple bindings, focusing on overcoming the most basic integration issues and exposing a reasonable subset of the deck.gl API, we should be able to provide significantly increase adoption and value of deck.gl in these areas.
+
+Offering such an integration clearly makes sense, as long as we can do it with low effort and maintenance cost.
+
+
+### Leveraging the deck.gl JSON API
```

Comment:
@ibgreen could you link the JSON API doc/RFC here for easy reference?

My understanding is the deck.gl JSON interface is flexible enough that we don't need frequent updates in accordance with the deck.gl APIs. However, to ensure the generated JSON configs match with the latest deck.gl APIs, developers would still need to follow closely with the API changes and test end-to-end from Python to JSON to JS, which seems to beat the purpose of decoupling Python and JS?

Reply:
> could you link the JSON API doc/RFC here for easy reference?

Done

> However, to ensure the generated JSON configs match with the latest deck.gl APIs, developers would still need to follow closely with the API changes and test end-to-end from Python to JSON to JS, which seems to beat the purpose of decoupling Python and JS?

Thanks for pointing this out. I added a bunch of text around completeness and stability of the JSON API.

For stability I feel it comes down to versioning. Our core libraries are carefully versioned. I believe that if we can apply the same versioning to the python module that we do to  our JavaScript module (not sure if Python uses semver), it will actually be good that the Python API evolves with the deck.gl API so that the latest deck.gl features can always be accessed through an upgrade of the python module.

External Python libraries might want to define a thick abstraction layer over deck.gl to hide our API and keep things stable, but I personally don't think that should be our job.